azure-cosmos/src/main/java/io/micronaut/azure/cosmos/client/CosmosClientConfiguration.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.azure.cosmos.client;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosClientBuilder;
+import io.micronaut.context.annotation.ConfigurationBuilder;
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.context.env.Environment;
+import io.micronaut.core.annotation.NonNull;
+
+/**
+ * The default Azure Cosmos configuration class.
+ *
+ * @author radovanradic
+ * @since 3.5.0
+ */
+@ConfigurationProperties(CosmosClientConfiguration.PREFIX)
+public final class CosmosClientConfiguration {
```

Comment:
Why final?

Reply:
Users can always create a custom configuration

Reply:
Ok, I am not sure what is preferred and consistent with the rest of the projects, leaving as is now.